### ISSUE TITLE
Why CanProduceMessages contract marked as @internal

### ISSUE THREAD

**Issue by joergbasedow** [mateusjunges/laravel-kafka#228]
Hi!

We very much like your package. Helps us a lot.

But we have a similar "issue" like [this](https://github.com/mateusjunges/laravel-kafka/issues/178).

We are using a code snippet like this in our project:

```
    private function buildProducer(): CanProduceMessages
    {
        $config = config('rola.kafka');
        $brokers = implode(',', $config['brokers']);
        $sasl = new Sasl(...$config['sasl']);

        return Kafka::publishOn(self::TOPIC, $brokers)
            ->withConfigOption('auto.offset.reset', 'earliest')
            ->withSasl($sasl);
    }
```

Unfortunately PhpStorm complains that `CanProduceMessages` is marked as internal.

Is there a different way to build a consumer with those options?

I could not think of a different class or interface to use as a return type.

Is `CanProduceMessages` really meant to only be used internally?

Any suggestions?

Thank you very much in advance,
Jörg

**Comment by lentex** [comment]
hey,
we are using a macro to achieve that:
```php
// in one of your Service Providers
public function boot() {
    $config = config('rola.kafka');
    $brokers = implode(',', $config['brokers']);
    $sasl = new Sasl(...$config['sasl']);

    Kafka::macro('producer', function (string $topic) use ($sasl, $brokers) {
            return $this->publishOn(self::TOPIC, $brokers)
                ->withSasl($sasl)
                ->withConfigOption('auto.offset.reset', 'earliest');
    });
}

// anywhere in your project
Kafka::producer()->withBodyKey('foo', ['foo' => 'bar'])->send();
```

Downside of macros is the missing autocompletion (at least I don't know how to tell PhpStorm how macros are resolved)

**Comment by joergbasedow** [comment]
Thanks, but then I guess I prefer to have it closer in the code to where I actually use the producer. Also being able to navigate the methods and having code completion in the IDE.

**Comment by mateusjunges** [comment]
Hmm I didn't had a use case where I needed that interface outside the package, so I marked it as internal. This seems like a fair example, will push a fix for that :+1:

**Comment by mateusjunges** [comment]
Just released v1.13.6 removing the internal annotation.
